stac_fastapi/api/stac_fastapi/api/models.py

```diff
@@ -121,9 +121,40 @@ class ItemCollectionUri(APIRequest):
         ),
     ] = attr.ib(default=10)
     bbox: Optional[BBox] = attr.ib(default=None, converter=_bbox_converter)
-    datetime: Optional[DateTimeType] = attr.ib(
-        default=None, converter=_datetime_converter
-    )
+    datetime: Annotated[
+        Optional[str],
+        Query(
+            description="""Only return items that have a temporal property that intersects this value.\n
+Either a date-time or an interval, open or closed. Date and time expressions adhere to RFC 3339. Open intervals are expressed using double-dots.""",  # noqa: E501
+            openapi_examples={
+                "datetime": {"value": "2018-02-12T23:20:50Z"},
+                "closed-interval": {"value": "2018-02-12T00:00:00Z/2018-03-18T12:31:12Z"},
+                "open-interval-from": {"value": "2018-02-12T00:00:00Z/.."},
+                "open-interval-to": {"value": "../2018-03-18T12:31:12Z"},
+            },
+        ),
+    ] = attr.ib(default=None)
+
+    @datetime.validator
+    def validate_datetime(self, attribute, value):
+        """Validate Datetime."""
+        _ = str_to_interval(value)
+
+    @property
+    def start_date(self) -> Optional[dt]:
+        """Start Date."""
+        if self.datetime is None:
+            return self.datetime
+        interval = str_to_interval(self.datetime)
+        return interval if isinstance(interval, dt) else interval[0]
+
+    @property
+    def end_date(self) -> Optional[dt]:
+        """End Date."""
+        if self.datetime is None:
+            return self.datetime
+        interval = str_to_interval(self.datetime)
+        return interval[1] if isinstance(interval, tuple) else None
```

Comment:
Feels funny that these conditionals are inverted:

```suggestion
        return interval[0] if isinstance(interval, tuple) else interval

    @property
    def end_date(self) -> Optional[dt]:
        """End Date."""
        if self.datetime is None:
            return self.datetime
        interval = str_to_interval(self.datetime)
        return interval[1] if isinstance(interval, tuple) else None
```

Also, is `None` correct for end date, instead of just the single datetime? E.g. this would break with how things are currently written, but seems like a reasonable thing to do?

```python
query = ItemCollectionUri(datetime="2025-01-14T16:00:00Z")
if query.start_date >= item.datetime and query.end_date <= item.datetime:
   ...
```

Reply:
good catch, will add some tests 🙏

Reply:
> Also, is None correct for end date, instead of just the single datetime?

maybe we could add another property `is_interval()`

Reply:
FYI: we don't use start/end dates in stac-fastapi-pgstac, I've added those to match with https://github.com/stac-utils/stac-pydantic/blob/main/stac_pydantic/api/search.py#L51-L58 which will also return `None` for a single datetime 

maybe we don't really need this anyway

Reply:
Yeah, I think an `.contains_datetime` method that handles the interval and non-interval cases would be safter than exposing derived `start_datetime` and `end_datetime`.

Reply:
what should `contains_datetime` return ?

Reply:
`bool | None` I think?

```python
def contains_datetime(self, datetime: dt) -> bool | None:
    if self.datetime is None:
        return None
    if self.datetime_is_interval():
        start, end = self.split_datetime()  # psuedocode
        return datetime >= start and datetime <= end
    else:
        return self.datetime == datetime
```

As I write this, just wanted to check we're handling no-timezone-datetimes correctly too (i.e. forcing them to UTC)?